arch/arm/core/aarch32/mpu/arm_mpu_v8_internal.h

```diff
@@ -28,28 +28,124 @@ struct dynamic_region_info {
  * regions may be configured.
  */
 static struct dynamic_region_info dyn_reg_info[MPU_DYNAMIC_REGION_AREAS_NUM];
+#if defined(CONFIG_CPU_CORTEX_M23) || defined(CONFIG_CPU_CORTEX_M33) || \
+	defined(CONFIG_CPU_CORTEX_M55)
+static inline void mpu_set_mair0(uint32_t mair0)
+{
+	MPU->MAIR0 = mair0;
+}
+
+static inline void mpu_set_rnr(uint32_t rnr)
+{
+	MPU->RNR = rnr;
+}
+
+static inline void mpu_set_rbar(uint32_t rbar)
+{
+	MPU->RBAR = rbar;
+}
 
+static inline uint32_t mpu_get_rbar(void)
+{
+	return MPU->RBAR;
+}
+
+static inline void mpu_set_rlar(uint32_t rlar)
+{
+	MPU->RLAR = rlar;
+}
+
+static inline uint32_t mpu_get_rlar(void)
+{
+	return MPU->RLAR;
+}
+
+static inline uint8_t mpu_get_num_regions(void)
+{
+	uint32_t type = MPU->TYPE;
+
+	type = (type & MPU_TYPE_DREGION_Msk) >> MPU_TYPE_DREGION_Pos;
+
+	return (uint8_t)type;
+}
+
+static inline void mpu_clear_region(uint32_t rnr)
+{
+	ARM_MPU_ClrRegion(rnr);
+}
+
+#elif defined(CONFIG_AARCH32_ARMV8_R)
+static inline void mpu_set_mair0(uint32_t mair0)
+{
+	write_mair0(mair0);
+	__DSB();
+	__ISB();
```

Comment:
Doesn't DSB already subsume an instruction barrier?  Seems weird to actually need both.

Reply:
AFAIK, `DSB` has no effect on pre-fetching of instructions and `ISB` will flush the pipelines to fetch next instructions from cache or memory.